source/Nevermore.Tests/CommandParameterValuesFixture.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using FluentAssertions;
+using Microsoft.Data.SqlClient;
+using NUnit.Framework;
+
+namespace Nevermore.Tests
+{
+    public class CommandParameterValuesFixture
+    {
+        [Test]
+        public void ShouldReplaceParametersWithExactMatch()
+        {
+            var parameters = new CommandParameterValues();
+            parameters.Add("someId", new[] { "id1", "id2" });
+            parameters.Add("someIdentifier", "value");
+            var command = new SqlCommand("SELECT * FROM [Table] WHERE [Id] IN @someId AND [OtherId] = @someIdentifier");
+
+            parameters.ContributeTo(command);
+
+            command.CommandText.Should().Be("SELECT * FROM [Table] WHERE [Id] IN (@someId_0, @someId_1) AND [OtherId] = @someIdentifier");
+            command.Parameters["someId_0"].Value.Should().Be("id1");
+            command.Parameters["someId_1"].Value.Should().Be("id2");
+            command.Parameters["someIdentifier"].Value.Should().Be("value");
+        }
+        
+        [Test]
+        public void ShouldReplaceParametersWithExactMatchEndOfQuery()
+        {
+            var parameters = new CommandParameterValues();
+            parameters.Add("someId", "value");
+            parameters.Add("someIdentifier", new[] { "id1", "id2" });
+            var command = new SqlCommand("SELECT * FROM [Table] WHERE [Id] = @someId AND [OtherId] IN @someIdentifier");
+
+            parameters.ContributeTo(command);
+
+            command.CommandText.Should().Be("SELECT * FROM [Table] WHERE [Id] = @someId AND [OtherId] IN (@someIdentifier_0, @someIdentifier_1)");
+            command.Parameters["someId"].Value.Should().Be("value");
+            command.Parameters["someIdentifier_0"].Value.Should().Be("id1");
+            command.Parameters["someIdentifier_1"].Value.Should().Be("id2");
+        }
+
+        [TestCase("@")]
```

Comment:
I think it _might_ be cleaner if you just put the whole identifier  as the test case i.e. `@someId#entifier`